mlrun/api/api/endpoints/feature_sets.py

```diff
@@ -32,16 +32,39 @@ def create_feature_set(
     )
 
 
-@router.put("/projects/{project}/feature_sets/{name}/references/{reference}")
-def update_feature_set(
+@router.put(
+    "/projects/{project}/feature_sets/{name}/references/{reference}",
+    response_model=schemas.FeatureSet,
+)
+def store_feature_set(
+    project: str,
+    name: str,
+    feature_set: schemas.FeatureSet,
+    reference: str,
+    versioned: bool = True,
+    db_session: Session = Depends(deps.get_db_session),
+):
+    tag, uid = parse_reference(reference)
+    uid = get_db().store_feature_set(
+        db_session, project, name, feature_set, tag, uid, versioned
+    )
+
+    return get_db().get_feature_set(db_session, project, name, uid=uid,)
+
+
+@router.patch("/projects/{project}/feature_sets/{name}/references/{reference}")
+def patch_feature_set(
     project: str,
     name: str,
-    feature_set_update: schemas.FeatureSetUpdate,
+    feature_set_update: dict,
     reference: str,
+    additive: bool = False,
```

Comment:
ok. Since the merge strategy may be useful for other PATCH operations as well, I've placed it in the general object scheams, not in the feature-set specific file.